### PR TITLE
Fixed #32543 -- Added search_help_text to ModelAdmin.

### PR DESCRIPTION
**Pull request by hramezani**
https://code.djangoproject.com/ticket/32543
Continue to https://github.com/django/django/pull/14117

### CODE REVIEW

File: tests/admin_changelist/tests.py [hunk 1]
```diff
@@ -1328,6 +1328,30 @@ def test_object_tools_displayed_no_add_permission(self):
...
+    def test_search_help_text(self):
+        superuser = self._create_superuser('superuser')
+        m = BandAdmin(Band, custom_site)
+
+        # Model admin with `search_fields` and without `search_help_text`
+        m.search_fields = ['name']
+        request = self._mocked_authenticated_request('/band/', superuser)
+        response = m.changelist_view(request)
+        self.assertIsNone(response.context_data['cl'].search_help_text)
+        self.assertNotIn(
```
Comment (felixxm) [inline tests/admin_changelist/tests.py:1340]: Use `assertContains()/assertNotContains()` instead.

**Review by felixxm — COMMENTED** [review]
@hramezani Thanks for this patch :+1:

File: django/contrib/admin/templates/admin/search_form.html [hunk 2]
```diff
@@ -11,6 +11,12 @@
 {% for pair in cl.params.items %}
     {% if pair.0 != search_var %}<input type="hidden" name="{{ pair.0 }}" value="{{ pair.1 }}">{% endif %}
 {% endfor %}
+{% if cl.search_help_text %}
+<br class="clear">
+<small class="quiet" style="margin-left: 23px;">
```
[the remark this answers is not in the document]
Comment (hramezani) [reply django/contrib/admin/templates/admin/search_form.html]: I added `word-break: break-word;` to break the words.

About the layout problem, I don't know how to fix it. Do you have any clue?

**Comment by hramezani** [comment]
Thanks @felixxm for the review. I've addressed all the comments except than the layout problem.

File: django/contrib/admin/templates/admin/search_form.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (hramezani) [reply django/contrib/admin/templates/admin/search_form.html]: I moved search_help_text block to the outside of `div` and I think it got fiexed.

**Review by felixxm — COMMENTED** [review]
@hramezani Thanks for updates :+1:

**Pull request merged**